core/templates/components/checkpoint-celebration-modal/checkpoint-celebration-modal.component.ts

```diff
@@ -106,17 +104,17 @@ export class CheckpointCelebrationModalComponent implements OnInit, OnDestroy {
       );
     this.readOnlyExplorationBackendApiService
       .fetchExplorationAsync(this.explorationId, null)
-      .then(response => {
-        this.exploration = response.exploration;
+      .then(exploration => {
+        this.exploration = exploration;
         this.hasViewedLessonInfoOnce =
-          response.has_viewed_lesson_info_modal_once;
+          exploration.metadata.has_viewed_lesson_info_modal_once;
         this.mostRecentlyReachedCheckpointStateName =
-          response.most_recently_reached_checkpoint_state_name;
+          exploration.metadata.most_recently_reached_checkpoint_state_name;
 
         this.orderedCheckpointList =
           this.checkpointCelebrationUtilityService.getStateListForCheckpointMessages(
-            this.exploration.states,
-            this.exploration.init_state_name
+            exploration.getStates().toBackendDict(),
```

Comment:
Why do we need to call toBackendDict() here? Better to pass in domain objects if possible between frontend functions, rather than dicts.